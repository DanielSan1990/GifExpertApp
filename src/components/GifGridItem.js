import React from 'react';
import 'animate.css';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
 
    return(
      
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt ={title}></img>
            <p>{title}</p>
        </div>
    )
};

GifGridItem.prototypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}