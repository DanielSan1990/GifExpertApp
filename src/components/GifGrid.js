import React from "react";
import { useFetchGifg } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {

  const { data, loading } = useFetchGifg(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && <p className="animate__animated animate__flash">Cargando...</p>}
      <div className="card-grid">
        {
          //Desatruturamos el array images para obtener el id y el title
          data.map((img) => {
            return <GifGridItem key={img.id} {...img} />;
          })
        }
      </div>
    </>
  );
};
