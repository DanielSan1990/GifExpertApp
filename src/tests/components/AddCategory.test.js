import React from "react";
import '@testing-library/jest-dom';
;

import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Pruebas AddCategory", () => {

  const setCategories = jest.fn();
  let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

  test("debe mostrar el componente corectamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");

    const value = "Hola mundo";

    input.simulate("change", {
      target: {
        value: value,
      },
    });
  });

  test('no debe de postear la informacion con submit', () => {

    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    expect( setCategories ).not.toHaveBeenCalled()

  });

  test('debe de postear la informacion con submit', () => {

    const value = 'Hola Mundo'

    wrapper.find( 'input' ).simulate('change', {target: {value}})

    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect( setCategories ).toHaveBeenCalled();

    expect( wrapper.find('input').prop('value')).toBe('')

  });
  
});
