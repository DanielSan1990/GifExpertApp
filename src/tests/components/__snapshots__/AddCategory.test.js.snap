// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas AddCategory debe mostrar el componente corectamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
